src/actions/index.js

```diff
@@ -1,42 +1,43 @@
+import { createAction } from 'redux-actions';
+import { Actions } from '../constants';
+
+
 /**
  * this variable holds unique incremental ID for every todo item.
  */
 let nextTodoId = 0;
 
-/**
- * this function returns an object to add a new item to todo list.
- */
-export const addTodo = (text) => {
-  return {
-    type: 'ADD_TODO',
-    id: nextTodoId++,
-    text
+const payload = {
+  getAddTodo: (text) => {
+    return {
+      id: nextTodoId++,
+      text
+    };
+  },
+  getMoveTodo: (sourceTodo, targetTodo) => {
+    return {
+      id: sourceTodo.id,
+      text: sourceTodo.text,
+      completed: sourceTodo.completed,
+      target_id: targetTodo.id
+    }
+  },
+  getVisibilityFilter: (filter) => {
```

Comment:
Yes, but i am modifying default arguments so as to create a specific object, like in the case of moveTodo. Also, each todo needs an unique id. So should i just have default arguments in the payload instead?